### ISSUE TITLE
CPHelper freezing

### ISSUE THREAD

**Issue by lcreed** [ToddAbbaticchio/FFXI-Addons#3]
I'm trying to use cphelper to auto buy scythes at the Windy Woods npc.  I loaded it and typed //ch buy but it went into a weird loop and froze my char in place, even after unloading the addon.  Any ideas on what i'm doing wrong?

![image](https://github.com/ToddAbbaticchio/FFXI-Addons/assets/734957/055536c8-1c36-456c-958a-7e45ca38c859)

**Comment by ToddAbbaticchio** [comment]
Based on your screenshot I've got a pretty good idea, but can't say for
sure because I'm not actively subbed anymore. I can ping a friend that
still plays and maybe have him get me some more info, but here's my initial
thoughts:

Lines 56-69 of cpHelper.lua are basically getting the list of all NPCs
within range, then iterating through the list and trying to match by name
to find the Player's and the CPVendor's index in the mob list. From then on
it uses that index ID to inject packets *telling* the ffxi server that you
went through the whole process of talking to npc, selecting the option to
buy things, selecting the item, and finalizing the purchase. For some
reason it isn't finding a match for the CPVendor. Maybe you're standing
really far away from them? Maybe they changed his name and it isn't
"Harara, W.W." anymore? Maybe something else.

I'd advise trying it in Windurst Waters. That's by far the best zone anyway
because you can go full auto mode - by that I mean there is a spot you can
stand where you're within 6 yalms of both the CP Vendor and a regional
vendor that will buy the scythes from you. As long as you're also using the
SellNPC addon, CPHelper will add the scythe to the autosell list and talk
to the regional vendor, selling all of them and repeat the process until
you're out of CP.

Let me know if that doesn't work - I've lowkey been thinking of resubbing
anyway. Maybe this'll be the push :)

**Comment by lcreed** [comment]
It did work perfectly in Windy Waters.  The NPC names are the same so
thinking maybe the id of the Woods NPC changed.  I was reading somewhere
that other lua's are broken due to the id being incremented by one.  I'm
not sure how to get an NPC id or I could change the lua easily enough.

So grateful you wrote this.  Saved me HOURS of time.  Thank you!

**Comment by ToddAbbaticchio** [comment]
Glad to hear it's working for you in W.Waters :)  And out of curiosity -
what server do you play on? Have you been playing since way back, or are
you a recent joiner? And how did you find CPHelper? If just random
google searching, that's wild.  All my github repositories were private
until about 2 weeks ago, I *just* opened them up.  Kind of cool that people
beyond my immediate circle of friends are already finding and using some of
it.

**Comment by lcreed** [comment]
I play on Bahamut now but originally from Shiva.  Recent returnee during
the last free play.  It's like smoking.  Long periods where you don't and
pretty soon you're hitting the crack pipe again.

Luckily I keep all my stuff backed up in repos on github.  Its always the
first place I search when I'm looking for an easy button to avoid a grindy
thing like buying from one npc and selling to another.  I've been looking
for something that will auto do my mog garden tasks forever!  I just
randomly searched for conquest points and ffxi on github and found your
lua.   Trying to farm up gil to buy beits so selling the CP helped a little.

**Comment by ToddAbbaticchio** [comment]
Yeah - things got craaaaaazy after they added the high level bats down in
the basement of King Ranperre's Tomb. Towards the end of my last stretch
playing I started to realize that I enjoyed the ability to whip up some
code that made a bad part of the game not so bad almost as much / if not
more than playing most of the time.  Sort of developed an attitude of
"Well, if they ban me for this I'm fine with that. I'd rather not play if I
can't do it anyway." After AutoFite, there was a period of time where some
friends and I would just setup in Tomb on bats every night before bed and
let it run until the next day when we got off work. After we all had all
the jobs we cared about at ML50 and we could focus on party comps for
optimum xp/hr instead of getting levels we'd pull north of 7 million xp /
hr if there wasn't another party on top of us stealing our bats.  It led to
some pretty silly piles of CP, and then after CPHelper silly piles of gil.
XD  I'm honestly shocked it hasn't been nerfed or stopped yet.


[image: image.png]
[image: image.png]

**Comment by lcreed** [comment]
Oh, you'll have to tell me more about autofite.  I run on linux so can't
use ezfarm or whatever everyone else is using these days.  Not worth
installing on a Windows vm to me.  I wouldn't mind waking up with some ep
levels on jobs i never get to play much anymore.

**Comment by ToddAbbaticchio** [comment]
Ahh, yeah.  Feel free to check it out, it's also in my ffxi addons github
repo.  The tl;dr is that you can define a set of rules based on the job
you're currently on.  You turn on autoFite and it behaves how you told it.
There's an 'assist mode' and a 'pull mode', typically one person is running
in pull mode and the rest of the party is running in assist mode with them
as the assist target.  The JobVars tables you create let you specify buffs
it should maintain, the WS it should spam and at what TP threshold, has
handling to do an alternate weaponskill as often as required to maintain
aftermath effects if desired, can auto-react to specified events with
actions, (if someone uses weaponskill x on the target, you close skillchain
with weaponskill y.  or if someone makes darkness on the target, you magic
burst with spell x.  Or if the target uses buff x on itself, you respond
with dispel. etc  it's totally customizable so you can do whatever you want
really.)  lots of stuff really.  It started out pretty simple and just
sorta grew. Had ~10 people using it for ~18 hours a day for almost a year.
Lots of feedback / ideas for new features and whatnot.

If you get around to checking it out and have questions feel free to yell.
Should work just fine on linux as it's a pure lua/addon solution rather
than a third party tool like ezfarm.

**Comment by lcreed** [comment]
Nice!  I'll play with it!  Been using trust but its not letting me approach
pull on drg/sam.
